### PR TITLE
HHH-9392 : SQLGrammarException while executing a entity graph with subgraphs

### PR DESCRIPTION
**Pull request by gbadner**
Proposed fix for HHH-9392.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java [hunk 1]
```diff
@@ -403,7 +403,7 @@ public void setOrigin(FromElement origin, boolean manyToMany) {
 				ASTUtil.appendSibling( origin, this );
 			}
 			else {
-				if ( !getWalker().isInFrom() && !getWalker().isInSelect() ) {
+				if ( !getWalker().isInFrom() && !getWalker().isInSelect() && !getWalker().isInEntityGraph()) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java]: I am not understanding this part of the change.   If we are not setting the origin here, where do we?

Comment (gbadner) [inline hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java]: It's easier to see looking add at it in context: https://github.com/gbadner/hibernate-core/blob/HHH-9392-master/hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java#L406 .

Without the additional check (&& !getWalker().isInEntityGraph()), the FromElement was being added as a child to the FromClause; to fix, the FromElement needed to be added as a child to the origin.

Really, isInEntityGraph() is a hacky way of getting around the fact that isInFrom() is false when processing the EntityGraph.

I considered adding a protected setter for HqlSqlBaseWalker.inFrom that HqlSqlWalker could use when processing the EntityGraph, but that seemed hackier.

Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java]: ok, that sounds reasonable

**Comment by gbadner** [comment]
Pushed to master and 4.3.

**Pull request closed without merging**